harvester/cli.py

```diff
@@ -32,3 +35,85 @@ def shell(ctx: click.Context) -> None:
     """Run a bash shell inside the docker container."""
     # ruff: noqa: S605, S607
     os.system("bash")
+
+
+@main.command()
+@click.option(
+    "--crawl-name",
+    required=True,
+    type=str,
+    help="Name of crawl",
+)
+@click.option(
+    "--config-yaml-file",
+    required=True,
+    type=str,
+    help="Filepath of browsertrix config YAML. Can be a local filepath or an S3 "
+    "URI, e.g. s3://bucketname/crawl-config.yaml",
+)
+@click.option(
+    "--sitemap-from-date",
+    required=False,
+    default=None,
+    type=str,
+    help="YYYY-MM-DD string to filter websites modified after this date in sitemaps",
+)
+@click.option(
+    "--wacz-output-file",
```

Comment:
This was a concious choice to match other apps like [OAI harvester](https://github.com/MITLibraries/oai-pmh-harvester/blob/main/harvester/cli.py#L24-L30) and [Transmog](https://github.com/MITLibraries/transmogrifier/blob/main/transmogrifier/cli.py#L23-L34) which both use this `-output-file` or `input-file` convention.  Willing to change, but I think the consistency across apps may be desirable.

Reply:
Interesting, I agree on consistency so I think it's worth having some best practices about how we name filepaths vs file objects, both of which could be `input_file`, to minimize confusion. This might be minor 
 but I think it's worth updating other repos if that's what we decide. @jonavellecuerdo Your thoughts?

Reply:
Agreed.  To me, `filename` (if ever used) should likely not include the path of any kind.

It's sublte, but I think it's safe to say `file` is never used by itself (or at least probably shouldn't be).  This means you see things like `input-file` or `output_file`.  We _also_ see `filepath` or `input_filepath`, etc.  While it'd be nice to be consistent across those, I'm okay with `<something>-file` and `<something>-filepath` being mostly interchangeable. 

With the caveat usage should be **consistent inside of a project**.  It does feel off to have `input-file` right next to `output-filepath` as arguments in the same project.

Reply:
I like `<something>-file` and `<something>-filepath` as a convention and agree that consistency inside a project is paramount. After @jonavellecuerdo weighs in, where should we document this?

Reply:
Hmmm, good question.  It feels along the lines of a "python styleguide" and wonder if [this page](https://mitlibraries.github.io/guides/languages/python.html) in the Dev Docs wiki is a good option?  Feels as though we could add a section like "Variable and Constant Naming Conventions".  In fact, we could probably add some other things we've been honing in on there as well (e.g. explictness over brevity, convention over idiosyncratic, etc.)  

I would want to avoid re-writing a style guides and best practices elsewhere, but anything that feels specific to our team maybe?

Reply:
Yeah, that page is appropriate and in need of an update 🙂

Reply:
Hi @ghukill and @ehanson8, I agree with the convention noted above. 
* `<something>-file` and `<something>-filepath` being interchangeable;
* consistent variable naming per project;
* `<something>-file` and <something>-filepath` can refer to full file paths (e.g., `directory/test.xml`) or a file-like object.

Reply:
My apologies, I misread Graham's comment and wasn't clear in my own:

- Consistent variable naming in a project
- My **preference** would be `-file` refers to file objects and `-filepath` is used for filepaths to minimize confusion. However, if it's at least consistent in a repo and others don't feel strongly, we can stick with interchangeable since that's basically the status quo.

Reply:
Given Eric's [comment](https://github.com/MITLibraries/browsertrix-harvester/pull/11#discussion_r1346368192) we might also want to differentiate between CLI arguments and variable names.

For example, I'm unsure how or when we'd pass a file **object** as a CLI command argument.

At the CLI level, I'd imagine we might want naming conventions of files vs directories, e.g. `--foo-file` vs `--foo-directory`.

But agreed that _within_ a python program there is more nuance (like filepaths vs open file objects, etc.).

Reply:
How about this is an item for discussion at the DataEng team meeting next week?